fastly/erl.go

```diff
@@ -0,0 +1,342 @@
+	if err != nil {
+		return nil, err
+	}
+
+	var erl *ERL
+	if err := decodeBodyMap(resp.Body, &erl); err != nil {
+		return nil, err
+	}
+
+	return erl, nil
+}
+
+// ********************************************************************************************* //
+//
+// Function 3: Delete rate limiter
+// Note(s):
+//
+// Return type
+//   error
+
+// Input to the DeleteERL function
+type DeleteERLInput struct {
+	ServiceID      string `form:"service_id"`
+	ServiceVersion int    `form:"version"`
+	ERLID          string `form:"id"`
+}
+
+// Deletes the specified ERL
+func (c *Client) DeleteERL(i *DeleteERLInput) error {
+	if i.ServiceID == "" {
+		return ErrMissingServiceID
+	}
+	if i.ServiceVersion == 0 {
+		return ErrMissingServiceVersion
+	}
+	if i.ERLID == "" {
+		return ErrMissingID
+	}
+
+	path := fmt.Sprintf("/rate-limiters/%s", i.ERLID)
+	resp, err := c.Delete(path, nil)
+	if err != nil {
+		return err
+	}
+
+	var r *statusResp
+	if err := decodeBodyMap(resp.Body, &r); err != nil {
+		return err
+	}
+	if !r.Ok() {
+		return fmt.Errorf("not ok")
+	}
+
+	return nil
+}
+
+// ********************************************************************************************* //
+//
+// Function 4: Get rate limiter
+// Note(s):
+//
+// Return type
+//   error
+
+// Input to the GetERL function
+type GetERLInput struct {
+	ServiceID      string `form:"service_id"`
+	ServiceVersion int    `form:"version"`
+	ERLID          string `form:"id"`
+}
+
+// Gets the specified ERL
+func (c *Client) GetERL(i *GetERLInput) (*ERL, error) {
+	if i.ServiceID == "" {
+		return nil, ErrMissingServiceID
+	}
+	if i.ServiceVersion == 0 {
+		return nil, ErrMissingServiceVersion
+	}
+	if i.ERLID == "" {
+		return nil, ErrMissingID
+	}
+
+	path := fmt.Sprintf("/rate-limiters/%s", i.ERLID)
+	resp, err := c.Get(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var erl *ERL
+	if err := decodeBodyMap(resp.Body, &erl); err != nil {
+		return nil, err
+	}
+
+	return erl, nil
+}
+
+// ********************************************************************************************* //
+//
+// Function 5: Update rate limiter
+// Note(s):
+//
+// Return type
+//   *ERL
+
+// Input to the UpdateERL function
+type UpdateERLInput struct {
+	ServiceID          string               `json:"-"`
+	ServiceVersion     int                  `json:"-"`
+	ID                 string               `json:"id"`
+	Action             ERLActionEnum        `json:"action,omitempty"`
+	ClientKey          []string             `json:"client_key,omitempty"`
+	HttpMethods        []ERLHttpMethodsEnum `json:"http_methods,omitempty"`
+	Name               string               `json:"name,omitempty"`
+	PenaltyBoxDuration int                  `json:"penalty_box_duration,omitempty"`
+	Response           *ERLResponseType     `json:"response,omitempty"`
+	RpsLimit           int                  `json:"rps_limit,omitempty"`
+	WindowSize         ERLWindowSizeEnum    `json:"window_size,omitempty"`
```

Comment:
Sorted and `Enum` suffix removed...

```suggestion
	Action             ERLAction        `json:"action,omitempty"`
	ClientKey          []string             `json:"client_key,omitempty"`
	HttpMethods        []ERLHttpMethods `json:"http_methods,omitempty"`
	ID                 string               `json:"id"`
	Name               string               `json:"name,omitempty"`
	PenaltyBoxDuration int                  `json:"penalty_box_duration,omitempty"`
	Response           *ERLResponseType     `json:"response,omitempty"`
	RpsLimit           int                  `json:"rps_limit,omitempty"`
	ServiceID          string               `json:"-"`
	ServiceVersion     int                  `json:"-"`
	WindowSize         ERLWindowSize    `json:"window_size,omitempty"`
```